Storage/SQL/BrowserDB.swift

```diff
@@ -206,9 +206,11 @@ open class BrowserDB {
         }
     }
 
-    func runQueryUnsafe<T>(_ sql: String, args: Args?, factory: @escaping (SDRow) -> T) -> Deferred<Maybe<Cursor<T>>> {
-        return withConnection { connection -> Cursor<T> in
-            connection.executeQueryUnsafe(sql, factory: factory, withArgs: args)
+    func runQueryUnsafe<T, U>(_ sql: String, args: Args?, factory: @escaping (SDRow) -> T, block: @escaping (Cursor<T>) -> U) -> Deferred<Maybe<U>> {
+        return withConnection { connection -> U in
+            let cursor = connection.executeQueryUnsafe(sql, factory: factory, withArgs: args)
```

Comment:
I think `executeQueryUnsafe`'s cursor implementation is flawed: its init touches every row in the cursor to get a count, because `Cursor`'s `makeIterator` implementation uses the count and subscripting. This is all kinds of silly.

I suggest a quick follow-up to define a better `makeIterator` (and probably other) method implementation for `LiveSQLiteCursor` so that it doesn't need to consume the whole cursor immediately.